docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -0,0 +1,461 @@
+		{
+			label: 'Guides',
+			items: [
+				{
+					label: 'Components',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
+				},
+			],
+		},
+	]}
+/>
+
+## Autogenerated groups
+
+Any group (or sub-group) can be automatically generated, and will be sorted alphabetically by filename. This is helpful when you do not want to manually enter each sidebar item in a group.
+
+To create an autogenerated group, set `autogenerate` to an object specifying the `directory`. For example, given the following file structure:
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - guides/
+        - components.md
+        - i18n.md
+        - advanced/
+          - project-structure.md
+
+</FileTree>
+
+And the following configuration:
+
+```js
+starlight({
+	sidebar: [
+		{
+			label: 'Guides',
+			// Autogenerate a group of links for the 'guides' directory.
+			autogenerate: { directory: 'guides' },
+		},
+	],
+});
+```
+
+The following sidebar will be generated:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+				{
+					label: 'advanced',
+					items: [
+						{ label: 'Project Structure', link: '/guides/project-structure/' },
+					],
+				},
+			],
+		},
+	]}
+/>
+
+### Customizing autogenerated links in frontmatter
+
+Use the [`sidebar` frontmatter field](/reference/frontmatter/#sidebar) to customize autogenerated links.
+
+This field allows you to set a [custom label](/reference/frontmatter/#label) or add a [badge](/reference/frontmatter/#badge) for a page link, to [hide](/reference/frontmatter/#hidden) a link from the sidebar or to define a [custom order](/reference/frontmatter/#order) for all page links.
+
+```md
+---
+title: My page
+sidebar:
+  # Set a custom label for the link
+  label: Custom sidebar label
+  # Show the link in the sidebar
+  hidden: false
+  # Set a custom order for the link (lower numbers are displayed higher up)
+  order: 2
+  # Add a badge to the link
+  badge:
+    text: Deprecated
+    variant: caution
+---
+```
+
+A sidebar including an autogenerated group for the `guides` directory containing a page with the above frontmatter will generate the following sidebar:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'Guides',
+			items: [
+				{ label: 'A page', link: '#' },
+				{
+					label: 'Custom sidebar label',
+					link: '#',
+					badge: { text: 'Deprecated', variant: 'caution' },
+				},
+				{ label: 'Another page', link: '#' },
+			],
+		},
+	]}
+/>
+
+:::note
+The `sidebar` frontmatter configuration is only used for autogenerated links and will be ignored for manually defined links.
+:::
+
+## Internationalization
+
+Manually defined link and group labels can be translated using the `translations` property to define a set of labels for each supported language while the `label` property will act as the default locale.
```

Comment:
I think we can split this into two sentences for slightly more punchiness and clarity:

```suggestion
Use the `translations` property on link and group entries to translate the link or group label for each supported language.
The `label` property will be used for the default locale and for languages without a translation.
```